File: app/src/main/java/com/vkumar/connect3/MainActivity.java
package com.vkumar.connect3;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.GridLayout;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    //Active Player - Yellow(0) Red (1) Blank = 2
    int activePlayer = 0;
    int[] gameState = {2,2,2,2,2,2,2,2,2};
    int[][] winningPositions = {{0,1,2},{3,4,5},{6,7,8},{0,3,6},{1,4,7},{2,5,8},{0,4,8},{2,4,6}};

    public void dropIn(View view){
        ImageView counter = (ImageView) view;
        int tagged = Integer.parseInt(counter.getTag().toString()) - 1;

        if (gameState[tagged] == 2){
            gameState[tagged] = activePlayer;
            counter.setTranslationY(-1000f);
            if (activePlayer == 0){
                counter.setImageResource(R.drawable.yellow);
                activePlayer = 1;
            } else {
                counter.setImageResource(R.drawable.red);
                activePlayer = 0;
            }
            counter.animate().translationYBy(1000f).setDuration(300);

            for (int[] winningPosition : winningPositions) {
                if (gameState[winningPosition[0]] == gameState[winningPosition[1]] &&
                        (gameState[winningPosition[1]] == gameState[winningPosition[2]]) &&
                        gameState[winningPosition[0]] != 2) {

                    TextView winnerMessage = (TextView)findViewById(R.id.textView);
                    String winner = "Red";

                    if (gameState[winningPosition[0]] == 0){
                        winner = "Yellow";
                    }
                    winnerMessage.setText("Player " + winner + " has WON!!!");

                    LinearLayout layout = (LinearLayout)findViewById(R.id.Layout);
                    layout.setVisibility(View.VISIBLE);
                }
            }
        } else {
            Toast.makeText(MainActivity.this,"Block is Already Taken!!!",Toast.LENGTH_LONG).show();
        }
    }
    public void playAgain(View view){
        LinearLayout layout = (LinearLayout)findViewById(R.id.Layout);
        layout.setVisibility(View.INVISIBLE);
        activePlayer = 0;
        for (int i=0;i < gameState.length;i++){
            gameState[i] = 2;
        }

        GridLayout gridLayout = (GridLayout)findViewById(R.id.gridLayout);

        for (int i=0; i< gridLayout.getChildCount(); i++) {
            ((ImageView) gridLayout.getChildAt(i)).setImageResource(0);
        }
    }
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
}

File: app/src/main/java/com/vkumar/connect3/MainActivity.kt
package com.vkumar.connect3

import android.os.Bundle
import android.view.View
import android.widget.*
import androidx.appcompat.app.AppCompatActivity
import com.microsoft.appcenter.AppCenter
import com.microsoft.appcenter.analytics.Analytics
import com.microsoft.appcenter.crashes.Crashes
import java.util.*

class MainActivity : AppCompatActivity() {
    //Active Player - Yellow(0) Red (1) Blank = 2
    private var activePlayer = 0
    private var gameState = intArrayOf(2, 2, 2, 2, 2, 2, 2, 2, 2)
    private val winningPositions = arrayOf(intArrayOf(0, 1, 2), intArrayOf(3, 4, 5), intArrayOf(6, 7, 8), intArrayOf(0, 3, 6), intArrayOf(1, 4, 7), intArrayOf(2, 5, 8), intArrayOf(0, 4, 8), intArrayOf(2, 4, 6))
    fun dropIn(view: View) {
        val gameStatus = findViewById<View>(R.id.tv_gameStatus) as TextView
        val counter = view as ImageView
        val tagged = counter.tag.toString().toInt() - 1
        if (gameState[tagged] == 2) {
            gameState[tagged] = activePlayer
            counter.translationY = -1000f
            activePlayer = if (activePlayer == 0) {
                counter.setImageResource(R.drawable.yellow)
                gameStatus.text = "It's Red Turn"
                1
            } else {
                counter.setImageResource(R.drawable.red)
                gameStatus.text = "It's Yellow Turn now"
                0
            }
            counter.animate().translationYBy(1000f).duration = 300
            val winnerMessage = findViewById<View>(R.id.textView) as TextView
            for (winningPosition in winningPositions) {
                if (gameState[winningPosition[0]] == gameState[winningPosition[1]] &&
                        gameState[winningPosition[1]] == gameState[winningPosition[2]] && gameState[winningPosition[0]] != 2) {
                    var winner = "Red"
                    if (gameState[winningPosition[0]] == 0) {
                        winner = "Yellow"
                    }
                    winnerMessage.text = String.format("Player %s has WON!!!", winner)
                    gameStatus.text = "Game is finished"
                    val layout = findViewById<View>(R.id.Layout) as LinearLayout
                    layout.visibility = View.VISIBLE
                }
            }
            // Check if still can be played
            if (2 !in gameState){
                winnerMessage.text = String.format("Game is finished in draw !!!")
                gameStatus.text = "Game is finished"
                val layout = findViewById<View>(R.id.Layout) as LinearLayout
                layout.visibility = View.VISIBLE
            }
        } else {
            Toast.makeText(this@MainActivity, "Block is Already Taken!!!", Toast.LENGTH_LONG).show()
        }
    }

    @Suppress("UNUSED_PARAMETER")
    fun playAgain(view: View?) {
        val layout = findViewById<View>(R.id.Layout) as LinearLayout
        layout.visibility = View.INVISIBLE
        val gameStatus = findViewById<View>(R.id.tv_gameStatus) as TextView
        gameStatus.text = "Yellow to Start"
        activePlayer = 0
        Arrays.fill(gameState, 2)
        val gridLayout = findViewById<View>(R.id.gridLayout) as androidx.gridlayout.widget.GridLayout
        for (i in 0 until gridLayout.childCount) {
            (gridLayout.getChildAt(i) as ImageView).setImageResource(0)
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        AppCenter.start(application, getString(R.string.appCentreSecret),
                Analytics::class.java, Crashes::class.java)
        setContentView(R.layout.activity_main)
    }
}